lib/Basic/LangOptions.cpp

```diff
@@ -105,6 +106,8 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
     addPlatformConditionValue("os", "watchOS");
   else if (triple.isiOS())
     addPlatformConditionValue("os", "iOS");
+  else if (triple.isAndroid())
+    addPlatformConditionValue("os", "Android");
```

Comment:
This change needs a test like `test/Parse/ConditionalCompilation/x64FreeBSDTarget.swift` for each supported triple.